cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -111,9 +111,11 @@ func CreateStaticPodFiles(manifestDir string, cfg *kubeadmapi.ClusterConfigurati
 		return err
 	}
 
+	kubernetesDir := filepath.Dir(manifestDir)
```

Comment:
Ok, this is not what I expect here. What we should do is to switch the `manifestDir` parameter of `CreateStaticPodFiles` to `kubernetesDir` and then use `GetStaticPodDirectory` to obtain the dir.
Unfortunately, that may require changes in the path manager interface and this is out of scope of this PR.
Hence, what we need to do here is to pass `kubeadmconstants.KubernetesDir` to `GetStaticPodSpecs` below, add a TODO for flipping `CreateStaticPodFiles` to taking `kubernetesDir` and open an issue with `help wanted` label.